docs/architecture/adr-005-subscription.md

```diff
@@ -0,0 +1,268 @@
+# ADR 5: Subscription
+
+## Changelog
+
+
+## Context
+
+Cosmos-SDK should allow developers to choose from a variety of payment models for their application. Subscriptions are one of the most popular payment models for Internet applications, so it is natural to provide this payment model as a module for sdk users.
+
+For fuller context around this issue: see [\#4642](https://github.com/cosmos/cosmos-sdk/issues/4642)
+
+## Decision
+
+Create Subscription module in SDK so that users can subscribe to various subscription services. Subscription collectors will send messages to collect subscription payments from users who have due subscriptions. Subscriptions are paid out directly from a user account. 
+
+Service providers can define valid periods their users can choose from (e.g.: 3 months, 6 months, 1 year) and their associated rates when they create the subscription service.
+
+If the user does not have enough funds to pay for the current period, the subscription is inactivated. The user can always refund the account and resubscribe.
+
+Users can define maximum limits on how many periods the subscription is valid for. If this limit elapses before a subscription renewal, the subscription is invalidated. This is to mitigate the common problem of unused subscriptions siphoning off funds from account without the user noticing. For example, a user can subscribe to a service with a period of 3 months with a limit of 4. This subscription will automatically expire after a year unless the user manually increases/removes the limit with a second `SubscribeMsg`.
+
+Users can also manually unsubscribe by submitting an `UnsubscribeMsg`
+
+## Status
+
+#### Proposed
+
+## Implementation Changes
+
+Introduce the following messages to `x/subscription` module:
+
+```go
+// CreateSubscriptionMsg allows Service Provider to create new subscription service.
+// Generate unique serviceID from hash(name|collector)
+type CreateSubscriptionMsg struct {
+    Name        string          // human-readable name for subscription service
+    Description string          // description for what the service provides
+    Amounts     []sdk.Coins     // amounts to be collected for each subscription period
```

Comment:
we should allow subscription providers to append new periods to the existing ones